app/controllers/application_controller.rb

```diff
@@ -36,7 +37,7 @@ class ApplicationController < ActionController::Base
     service_unavailable
   end
 
-  before_action :store_current_location, except: :raise_not_found, unless: :devise_controller?
+  before_action :store_referrer, except: :raise_not_found, if: :devise_controller?
```

Comment:
`devise_controller?` will not be true for Doorkeeper paths such as OAuth authorization (I think)

Reply:
Good point, though I'm not sure that's needed in the first place for OAuth, since OAuth should redirect you out of the app anyway.
What I'm more concerned about is external authentication.
I will try both.

Reply:
You’re likely to not be logged in when you get redirected to the oauth authorization page, so it redirects you to login. It’s important that login redirects you back or it will become impossible to authenticate to apps.